InProcess deployer fails to deploy

InProcess deployer fails to deploy the modules as the `module launcher` includes the `LocalAutoConfiguration` along with its classpath [here] (https://github.com/spring-cloud/spring-cloud-stream/blob/master/spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/utils/ClassloaderUtils.java#L61)

The exception while deployer using local in-process deployer is:

```
Caused by: java.lang.NoClassDefFoundError: org/springframework/cloud/dataflow/module/deployer/ModuleDeployer
	at java.lang.Class.getDeclaredMethods0(Native Method)
	at java.lang.Class.privateGetDeclaredMethods(Class.java:2688)
	at java.lang.Class.getDeclaredMethods(Class.java:1962)
	at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:609)
	at org.springframework.util.ReflectionUtils.doWithMethods(ReflectionUtils.java:521)
	at org.springframework.util.ReflectionUtils.doWithMethods(ReflectionUtils.java:507)
	at org.springframework.util.ReflectionUtils.getUniqueDeclaredMethods(ReflectionUtils.java:567)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.getTypeForFactoryMethod(AbstractAutowireCapableBeanFactory.java:683)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.determineTargetType(AbstractAutowireCapableBeanFactory.java:627)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.predictBeanType(AbstractAutowireCapableBeanFactory.java:597)
	at org.springframework.beans.factory.support.AbstractBeanFactory.isFactoryBean(AbstractBeanFactory.java:1445)
	at org.springframework.beans.factory.support.AbstractBeanFactory.isFactoryBean(AbstractBeanFactory.java:975)
	at org.springframework.boot.autoconfigure.condition.BeanTypeRegistry$OptimizedBeanTypeRegistry.addBeanTypeForNonAliasDefinition(BeanTypeRegistry.java:289)
	at org.springframework.boot.autoconfigure.condition.BeanTypeRegistry$OptimizedBeanTypeRegistry.addBeanType(BeanTypeRegistry.java:278)
	at org.springframework.boot.autoconfigure.condition.BeanTypeRegistry$OptimizedBeanTypeRegistry.getNamesForType(BeanTypeRegistry.java:259)
	at org.springframework.boot.autoconfigure.condition.OnBeanCondition.collectBeanNamesForType(OnBeanCondition.java:182)
	at org.springframework.boot.autoconfigure.condition.OnBeanCondition.getBeanNamesForType(OnBeanCondition.java:171)
	at org.springframework.boot.autoconfigure.condition.OnBeanCondition.getMatchingBeans(OnBeanCondition.java:139)
	at org.springframework.boot.autoconfigure.condition.OnBeanCondition.getMatchOutcome(OnBeanCondition.java:86)
	at org.springframework.boot.autoconfigure.condition.SpringBootCondition.matches(SpringBootCondition.java:47)
	... 23 more
```

--- a/spring-cloud-dataflow-admin-local/src/main/java/org/springframework/cloud/dataflow/admin/spi/local/LocalDeployerAutoConfiguration.java
+++ b/spring-cloud-dataflow-admin-local/src/main/java/org/springframework/cloud/dataflow/admin/spi/local/LocalDeployerAutoConfiguration.java
@@ -17,6 +17,7 @@
 package org.springframework.cloud.dataflow.admin.spi.local;
 
 import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
@@ -32,9 +33,11 @@ import org.springframework.context.annotation.Import;
  *
  * @author Eric Bottard
  * @author Josh Long
+ * @author Ilayaperumal Gopinathan
  */
 @Configuration
-public class LocalAutoConfiguration {
+@ConditionalOnClass(ModuleDeployer.class)
+public class LocalDeployerAutoConfiguration {
 
 	static final String LOCAL_DEPLOYER_PREFIX = "deployer.local";
 

--- a/spring-cloud-dataflow-admin-local/src/main/java/org/springframework/cloud/dataflow/admin/spi/local/OutOfProcessModuleDeployerProperties.java
+++ b/spring-cloud-dataflow-admin-local/src/main/java/org/springframework/cloud/dataflow/admin/spi/local/OutOfProcessModuleDeployerProperties.java
@@ -26,7 +26,7 @@ import org.springframework.boot.context.properties.ConfigurationProperties;
  *
  * @author Eric Bottard
  */
-@ConfigurationProperties(prefix = LocalAutoConfiguration.LOCAL_DEPLOYER_PREFIX)
+@ConfigurationProperties(prefix = LocalDeployerAutoConfiguration.LOCAL_DEPLOYER_PREFIX)
 public class OutOfProcessModuleDeployerProperties {
 
 	/**

--- a/spring-cloud-dataflow-admin-local/src/main/resources/META-INF/spring.factories
+++ b/spring-cloud-dataflow-admin-local/src/main/resources/META-INF/spring.factories
@@ -1,2 +1,2 @@
 org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
-  org.springframework.cloud.dataflow.admin.spi.local.LocalAutoConfiguration
+  org.springframework.cloud.dataflow.admin.spi.local.LocalDeployerAutoConfiguration

--- a/spring-cloud-dataflow-admin-local/src/test/java/org/springframework/cloud/dataflow/admin/spi/local/LocalConfigurationTests.java
+++ b/spring-cloud-dataflow-admin-local/src/test/java/org/springframework/cloud/dataflow/admin/spi/local/LocalConfigurationTests.java
@@ -27,7 +27,7 @@ import org.springframework.cloud.dataflow.admin.AdminApplication;
 import org.springframework.context.ConfigurableApplicationContext;
 
 /**
- * Tests for {@link LocalAutoConfiguration}.
+ * Tests for {@link LocalDeployerAutoConfiguration}.
  *
  * @author Janne Valkealahti
  * @author Eric Bottard

